Plots/Polygons/NCL_polyg_19.py

```diff
@@ -0,0 +1,200 @@
+import matplotlib.cm as cm
+import shapefile as shp
+import numpy as np
+
+import geocat.datafiles as gdf
+from geocat.viz import util as gvutil
+
+###############################################################################
+# Read in data:
+
+file1 = open(gdf.get("shape_files/tl_2017_us_state.dbf"), 'r')
+file2 = open(gdf.get("shape_files/tl_2017_us_state.shp"), 'r')
+file3 = open(gdf.get("shape_files/tl_2017_us_state.shx"), 'r')
+file4 = open(gdf.get("shape_files/tl_2017_us_state.prj"), 'r')
+
+# Open the shapefile
+us = shp.Reader(gdf.get("shape_files/tl_2017_us_state.dbf"))
+
+# Open the text file with the population data
+statepopulationfile = open(gdf.get("ascii_files/us_state_population.txt"), 'r')
+
+###############################################################################
+# Set colormap data and colormap bounds:
+
+colormap = colors.ListedColormap(['lightpink', 'blanchedalmond', 'palegoldenrod', 'powderblue', 'thistle',
+                                  'lightcoral', 'peru', 'dodgerblue', 'slateblue', 'firebrick', 'sienna', 'olivedrab',
+                                  'steelblue', 'navy'])
+
+colorbounds = [0, 1, 2.5, 3, 4, 5, 6, 7, 8, 9, 10, 12, 25, 38, 40]
+
+norm = colors.BoundaryNorm(colorbounds, colormap.N)
+
+###############################################################################
+# Define helper function to get the populations of each state
+
+
+def getStatePopulations(statepopulationfile):
+
+    nameandpopdict = {}
+    Lines = statepopulationfile.read().splitlines()
+    for line in Lines:
+        try:
+            nameandpop = line.split(" ")
+            name = nameandpop[0]
+            pop = (int)(nameandpop[-1])/1000000
+            nameandpopdict[name] = pop
+        except Exception as E:
+            print(E)
+            continue
+
+    return nameandpopdict
+
+###############################################################################
+# Define helper function to get the color of each state based on its population
+
+
+def findDivColor(colorbounds, pdata):
+
+    for x in range(len(colorbounds)):
+
+        if pdata >= colorbounds[len(colorbounds)-1]:
+            return colormap.colors[x-1]
+        if pdata >= colorbounds[x]:
+            continue
+        else:
+            # Index is 'x-1' because colorbounds is one item longer than colormap
+            return colormap.colors[x-1]
+
+###############################################################################
+# Define helper function to remove ticks from axes
+
+
+def removeTicks(axis):
+
+    axis.get_xaxis().set_visible(False)
+    axis.get_yaxis().set_visible(False)
+
+###############################################################################
+# Define helper function to plot and color each state
+
+
+def plotRegion(shape, axis, xlim):
+
+    # Plot each shape within a region (ex. mainland Alaska and all of it's surrounding Alaskan islands)
+    for i in range(len(shape.shape.parts)):
```

Comment:
I would recommend refactoring the `shape` argument to either `state` or `region` since you are specifying a particular state when you call this function. The code is confusing with calls like `shape.shape.parts`.